File: IT/Funzioni/C++/passByReference.cpp
/* file passByReference.cpp */

#include <iostream>

using std::cout;
using std::endl;

void resetArray(int arr[]) {
	for (int i = 0; i < 3; i++) {
		arr[i] = 0;
	}
}

int main()
{
	int arr[] = {1, 2, 3};

	for (int i = 0; i < 3; i++) {
		cout << arr[i] << " ";
	}

	cout << endl;

	resetArray(arr);

	for (int i = 0; i < 3; i++) {
		cout << arr[i] << " ";
	}

	cout << endl;
}

File: IT/1/Funzioni/Java/PassByReference.java
/* file PassByReference.java */

public class PassByReference {
	static void resetArray(int [] arr) {
		for (int i = 0; i < arr.length; i++) {
			arr[i] = 0;
		}
	}

	public static void main(String [] args) {
		int arr[] = {1, 2, 3};

		for (int n : arr) {
			System.out.print(n + " ");
		}

		System.out.print(System.lineSeparator());

		resetArray(arr);

		for (int n : arr) {
			System.out.print(n + " ");
		}

		System.out.print(System.lineSeparator());
	}
}

